app/src/main/java/com/infomaniak/mail/ui/main/newMessage/NewMessageFragment.kt

```diff
@@ -163,15 +171,73 @@ class NewMessageFragment : Fragment() {
     }
 
     private fun initDraftAndUi() {
-        newMessageViewModel.initDraftAndUi(newMessageActivityArgs)
-            .observe(viewLifecycleOwner) { isSuccess ->
-                if (isSuccess) {
-                    observeContacts()
-                    populateUiWithExistingDraftData()
-                } else {
-                    requireActivity().finish()
-                }
+        newMessageViewModel.initDraftAndUi(newMessageActivityArgs).observe(viewLifecycleOwner) { isSuccess ->
+            if (isSuccess) {
+                observeContacts()
+                populateUiWithExistingDraftData()
+                handleActionSend()
+            } else {
+                requireActivity().finish()
+            }
+        }
+    }
+
+    private fun handleActionSend() {
+        when (requireActivity().intent?.action) {
+            Intent.ACTION_SEND -> handleSingleSendIntent()
+            Intent.ACTION_SEND_MULTIPLE -> handleMultipleSendIntent()
+        }
+    }
+
+    /**
+     * Handle `Mailto` from [Intent.ACTION_VIEW] or [Intent.ACTION_SENDTO]
+     * Get [Intent.ACTION_VIEW] data with [MailTo] and [Intent.ACTION_SENDTO] with [Intent]
+     */
+    private fun handleMailTo() = with(binding) {
+        val intent = requireActivity().intent
+        intent?.data?.let { uri ->
+            if (!MailTo.isMailTo(uri)) return@with
+
+            val mailTo = MailTo.parse(uri)
+            val to = mailTo.to?.split(",")?.map { it.trim() } ?: emptyList()
+            val cc = mailTo.cc?.split(",") ?: intent.getStringArrayExtra(Intent.EXTRA_CC)?.toList() ?: emptyList()
+            val bcc = mailTo.bcc?.split(",") ?: intent.getStringArrayExtra(Intent.EXTRA_BCC)?.toList() ?: emptyList()
```

Comment:
Why is only the "to" trim?

Reply:
I agree, This is an oversight during my conflicts 😅